core/vm/evm.go

```diff
@@ -307,6 +307,11 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 	return ret, gas, err
 }
 
+// CallFromPrecompile invokes Call to execute the contract with the given input parameters.
+func (evm *EVM) CallFromPrecompile(caller common.Address, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+	return evm.Call(AccountRef(caller), addr, input, gas, value)
```

Comment:
Should we be converting `caller` to a `ContractRef` expected by `Call`, rather than an `AccountRef`? I assume there's an implicit conversion available, but may as well use what's expected I think.

Reply:
`ContractRef` is an interface implemented by `AccountRef`. Do you mean to use `Contract` which also implements `ContractRef` ?

Reply:
Doesn't matter either way.